x-pack/plugins/global_search_providers/public/providers/get_app_results.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import levenshtein from 'js-levenshtein';
+import { PublicAppInfo, PublicLegacyAppInfo } from 'src/core/public';
+import { GlobalSearchProviderResult } from '../../../global_search/public';
+
+export const getAppResults = (
+  term: string,
+  apps: Array<PublicAppInfo | PublicLegacyAppInfo>
+): GlobalSearchProviderResult[] => {
+  return (
+    apps
+      // remove disabled and chromeless app (to avoid returning the login page for example)
+      .filter((app) => app.status === 0 && (app.legacy === true || app.chromeless !== true))
+      .map((app) => ({ app, score: scoreApp(term, app) }))
+      .filter(({ score }) => score > 0)
+      .map(({ app, score }) => appToResult(app, score))
+  );
+};
+
+export const scoreApp = (term: string, { title }: PublicAppInfo | PublicLegacyAppInfo): number => {
+  term = term.toLowerCase();
+  title = title.toLowerCase();
+
+  // shortcuts to avoid calculating the distance when there is an exact match somewhere.
+  if (title === term) {
+    return 100;
+  }
+  if (title.startsWith(term)) {
+    return 90;
+  }
+  if (title.includes(term)) {
+    return 75;
+  }
+  const length = Math.max(term.length, title.length);
+  const distance = levenshtein(term, title);
+
+  // maximum lev distance is length, we compute the match ratio (lower distance is better)
+  const ratio = Math.floor((1 - distance / length) * 100);
+  if (ratio >= 60) {
+    return ratio;
+  }
```

Comment:
Would it make sense for all matches that reach this part to be ranked lower than the exact matches above? If so, we should be multiplying the ratio by 74 instead of 100 to lower the ceiling.

Reply:
If we do that, we will have all 'searchable' apps returned on every search. We can tweak/weight the final ratio differently if we want to, but we will still need a score floor behind which we are excluding results.

Just impacting the final score could look like something like

```
const ratio = Math.floor((1 - distance / length) * 100);
  if (ratio >= 60) {
    return ratio * 0.74;
  }
```

Reply:
Ah yes, that's more what I meant. Let's just keep as-is for now and we can tweak this once we have a UI to test this out with.